### PR TITLE
cargo-release: remove `rust` dependency except for building

### PR DESCRIPTION
**Pull request by rami3l**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

cc #124998

This PR applies the fix from #134064 to `cargo-release`.

### CODE REVIEW

**Comment by carlocab** [comment]
Looks like this also needs:
```diff
diff --git a/Formula/cargo-release.rb b/Formula/cargo-release.rb
index eb60ced7354..beb9965458a 100644
--- a/Formula/cargo-release.rb
+++ b/Formula/cargo-release.rb
@@ -16,11 +16,25 @@ class CargoRelease < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "d63c11ca314240c51f31f66765f8daa1dcfd8ba9a7feb01de8eff9ec21096da1"
   end
 
+  depends_on "pkg-config" => :build
   depends_on "rust" => :build
   depends_on "rustup-init" => :test
+  depends_on "libgit2"
+  depends_on "libssh2"
+  depends_on "openssl@1.1"
 
   def install
-    system "cargo", "install", *std_cargo_args
+    ENV["OPENSSL_DIR"] = Formula["openssl@1.1"].opt_prefix
+    ENV["OPENSSL_NO_VENDOR"] = "1"
+    system "cargo", "install", "--no-default-features", *std_cargo_args
+  end
+
+  def check_binary_linkage(binary, library)
+    binary.dynamically_linked_libraries.any? do |dll|
+      next false unless dll.start_with?(HOMEBREW_PREFIX.to_s)
+
+      File.realpath(dll) == File.realpath(library)
+    end
   end
 
   test do
@@ -35,5 +49,15 @@ class CargoRelease < Formula
     cd "hello_world" do
       assert_match "tag = true", shell_output("cargo release config 2>&1").chomp
     end
+
+    [
+      Formula["libgit2"].opt_lib/shared_library("libgit2"),
+      Formula["libssh2"].opt_lib/shared_library("libssh2"),
+      Formula["openssl@1.1"].opt_lib/shared_library("libssl"),
+      Formula["openssl@1.1"].opt_lib/shared_library("libcrypto"),
+    ].each do |library|
+      assert check_binary_linkage(bin/"cargo-release", library),
+             "No linkage with #{library.basename}! Cargo is likely using a vendored version."
+    end
   end
 end
```

Unsure about the exact library linkage, but this looks right. We may have to drop the check for linkage with `libssh2` and/or one of `libssl`/`libcrypto`.

**Review by carlocab — CHANGES_REQUESTED** [review]
Let's fix linkage with vendored libraries while we're here. Thanks!

**Comment by rami3l** [comment]
> ```diff
> +    [
> +      Formula["libgit2"].opt_lib/shared_library("libgit2"),
> +      Formula["libssh2"].opt_lib/shared_library("libssh2"),
> +      Formula["openssl@1.1"].opt_lib/shared_library("libssl"),
> +      Formula["openssl@1.1"].opt_lib/shared_library("libcrypto"),
> +    ].each do |library|
> +      assert check_binary_linkage(bin/"cargo-release", library),
> +             "No linkage with #{library.basename}! Cargo is likely using a vendored version."
> +    end
>    end
>  end
> ```
> 
> Unsure about the exact library linkage, but this looks right. We may have to drop the check for linkage with `libssh2` and/or one of `libssl`/`libcrypto`.

@carlocab Sorry, I'm not that familiar with what I'm supposed to do in this case...
That being said, now the assertion fails at `libgit2`, which means that I should remove `libgit2` in both dependencies and assertions and try again, right?

**Comment by rami3l** [comment]
Hmmm. This doesn't look right. If all default features are disabled, then `libgit2` should have been dynamically linked:


```toml
[features]
default = ["vendored-libgit2", "vendored-openssl"]
vendored-libgit2 = ["git2/vendored-libgit2"]
vendored-openssl = ["git2/vendored-openssl"]
```

https://github.com/crate-ci/cargo-release/blob/c8bc39d2cb0c821913869cfe7572e2a7fdb00bc3/Cargo.toml#L46-L58

File: Formula/cargo-release.rb [hunk 1]
```diff
@@ -16,16 +16,50 @@ class CargoRelease < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "d63c11ca314240c51f31f66765f8daa1dcfd8ba9a7feb01de8eff9ec21096da1"
   end
 
-  depends_on "rust" # uses `cargo` at runtime
+  depends_on "pkg-config" => :build
+  depends_on "rust" => :build
+  depends_on "rustup-init" => :test
+  depends_on "libgit2"
```
Comment (carlocab) [inline Formula/cargo-release.rb:22]:
```suggestion
  depends_on "libgit2@1.5"
```

File: Formula/cargo-release.rb [hunk 2]
```diff
@@ -16,16 +16,50 @@ class CargoRelease < Formula
...
+    ENV.prepend_path "PATH", HOMEBREW_CACHE/"cargo_cache/bin"
+    system "rustup", "default", "beta"
+
     system "cargo", "new", "hello_world", "--bin"
     cd "hello_world" do
       assert_match "tag = true", shell_output("cargo release config 2>&1").chomp
     end
+
+    [
+      Formula["libgit2"].opt_lib/shared_library("libgit2"),
```
Comment (carlocab) [inline Formula/cargo-release.rb:56]:
```suggestion
      Formula["libgit2@1.5"].opt_lib/shared_library("libgit2"),
```

**Review by carlocab — COMMENTED** [review]
This should fix the `libgit2` issue.

File: Formula/cargo-release.rb (hunk 1 above)
Comment (rami3l) [reply Formula/cargo-release.rb:22]: I added the following from `cargo-c`. Shall we wait and see what happens first?

```ruby
ENV["LIBGIT2_SYS_USE_PKG_CONFIG"] = "1"
ENV["LIBSSH2_SYS_USE_PKG_CONFIG"] = "1"
```

Comment (rami3l) [reply Formula/cargo-release.rb:22]: Yes, you're right! This has worked on #134068. I'll reflect these changes here right away.

Comment (rami3l) [reply Formula/cargo-release.rb:22]: Yes, you're right! This has worked on #134068. I'll reflect these changes here right away.

Comment (rami3l) [reply Formula/cargo-release.rb]: Done.

File: Formula/cargo-release.rb (hunk 2 above)
Comment (rami3l) [reply Formula/cargo-release.rb]: Done.

**Review by carlocab — APPROVED** [review]
Thanks!

**Comment by rami3l** [comment]
Oops. `"libssh2"` seems like a mistake. I'll remove it and retry.

**Review by carlocab — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request merged**